src/EditorFeatures/CSharpTest/EditAndContinue/StatementMatchingTests.cs

```diff
@@ -863,6 +863,323 @@ public void LocalFunctionDefinitions()
+        {
+            int d() { return 0; };
+        };
+
+        return c();
+    };
+
+    return b();
+}
+";
+
+            var matches = GetMethodMatches(src1, src2);
+            var actual = ToMatchingPairs(matches);
+
+            var expected = new MatchingPairs
+            {
+                { "int a()  {     int b()     {         int c()         {             int d() { return 0; };         };          return c();     };      return b(); }", "int a()  {     int b()     {         int c()         {             int d() { return 0; };         };          return c();     };      return b(); }" },
+                { "{     int b()     {         int c()         {             int d() { return 0; };         };          return c();     };      return b(); }", "{     int b()     {         int c()         {             int d() { return 0; };         };          return c();     };      return b(); }" },
+                { "int b()     {         int c()         {             int d() { return 0; };         };          return c();     }", "int b()     {         int c()         {             int d() { return 0; };         };          return c();     }" },
+                { "{         int c()         {             int d() { return 0; };         };          return c();     }", "{         int c()         {             int d() { return 0; };         };          return c();     }" },
+                { "int c()         {             int d() { return 0; };         }", "int c()         {             int d() { return 0; };         }" },
+                { "{             int d() { return 0; };         }", "{             int d() { return 0; };         }" },
+                { "int d() { return 0; }", "int d() { return 0; }" },
+                { "{ return 0; }", "{ return 0; }" },
+                { "return 0;", "return 0;" },
+                { "return c();", "return c();" },
+                { "return b();", "return b();" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void LocalFunctions6()
+        {
+            var src1 = @"
+int a() 
+{
+    int b()
+    {
+        int c()
+        {
+            int d() { return 0; };
+        };
+
+        return c();
+    };
+
+    return b();
+}
+";
+            var src2 = @"
+int a() 
+{
+    int b()
+    {
+        int c()
+        {
+            int d() { return H(0); };
+        };
+
+        return G(c());
+    };
+
+    return F(b());
+}
+";
+
+            var matches = GetMethodMatches(src1, src2);
+            var actual = ToMatchingPairs(matches);
+
+            var expected = new MatchingPairs
+            {
+                { "int a()  {     int b()     {         int c()         {             int d() { return 0; };         };          return c();     };      return b(); }", "int a()  {     int b()     {         int c()         {             int d() { return H(0); };         };          return G(c());     };      return F(b()); }" },
```

Comment:
Thanks! Let me try to remove extra lines from code snippets and add splits in the verifications.

---
In reply to: [132525353](https://github.com/dotnet/roslyn/pull/21396#discussion_r132525353) [](ancestors = 132525353)